models/src/node/crd/mod.rs

```diff
@@ -0,0 +1,100 @@
+pub mod v1;
+pub mod v2;
+
+pub use self::v2::{
+    BottlerocketShadow, BottlerocketShadowSpec, BottlerocketShadowState, BottlerocketShadowStatus,
+};
+
+use crate::node::{error, error::Error};
+
+use k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition;
+use kube::CustomResourceExt;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::fmt::Debug;
+
+lazy_static! {
+    static ref BOTTLEROCKETSHADOW_CRD_METHODS: Vec<fn() -> CustomResourceDefinition> = {
+        // A list of CRD methods for different BottlerocketShadow version
+        // The latest version should be added at the end of the vector.
+        let mut crd_methods = Vec::new();
+        crd_methods.push(v1::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods.push(v2::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods
+    };
+}
+
+pub trait Selector {
+    fn selector(&self) -> error::Result<BottlerocketShadowSelector>;
+}
+
+impl<T: kube::ResourceExt> Selector for T {
+    fn selector(&self) -> error::Result<BottlerocketShadowSelector> {
+        let node_owner = self
+            .meta()
+            .owner_references
+            .as_ref()
+            .ok_or(Error::MissingOwnerReference { name: self.name() })?
+            .first()
+            .ok_or(Error::MissingOwnerReference { name: self.name() })?;
+
+        Ok(BottlerocketShadowSelector {
+            node_name: node_owner.name.clone(),
+            node_uid: node_owner.uid.clone(),
+        })
+    }
+}
+
+/// Indicates the specific k8s node that BottlerocketShadow object is associated with.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BottlerocketShadowSelector {
+    pub node_name: String,
+    pub node_uid: String,
+}
+
+impl BottlerocketShadowSelector {
+    pub fn brs_resource_name(&self) -> String {
+        brs_name_from_node_name(&self.node_name)
+    }
+}
+
+impl fmt::Display for BottlerocketShadowSelector {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "({}, {})", self.node_name, self.node_uid)
+    }
+}
+
+pub fn brs_name_from_node_name(node_name: &str) -> String {
+    format!("brs-{}", node_name)
+}
+
+pub fn combine_crd(
+    latest_crd: CustomResourceDefinition,
+    old_crds: Vec<CustomResourceDefinition>,
+) -> CustomResourceDefinition {
+    // Combine all different versions of customer resources into one CustomResourceDefinition yaml
+    // Related open issue: https://github.com/kube-rs/kube-rs/issues/569
```

Comment:
We should elevate some of this comment to be a function docstring. I think it might be a good idea to also write a module-level docstring explaining the layout of the `node::crd` package, and why it's done this way to enable migrations.